crates/parry3d/examples/plane_intersection.rs

```diff
@@ -0,0 +1,156 @@
+            IntersectResult::Intersect(points) => {
+                draw_polyline(
+                    points
+                        .segments()
+                        .map(|s| (mquad_from_na(s.a), mquad_from_na(s.b)))
+                        .collect(),
+                    Color::new(0f32, 1f32, 0f32, 1f32),
+                );
+                set_default_camera();
+                draw_text("Intersection found!");
+            }
+            IntersectResult::Negative => {
+                set_default_camera();
+                draw_text("No intersection found, the shape is below the plane.");
+            }
+            IntersectResult::Positive => {
+                set_default_camera();
+                draw_text("No intersection found, the shape is above the plane.");
+            }
+        }
+        next_frame().await
+    }
+}
+
+fn mquad_mesh_from_points(trimesh: &(Vec<Point3<Real>>, Vec<[u32; 3]>), camera_pos: Vec3) -> Mesh {
+    let (points, indices) = trimesh;
+    // Transform the parry mesh into a mquad Mesh
+    let (mquad_points, mquad_indices) = (
+        points
+            .iter()
+            .map(|p| Vertex {
+                position: mquad_from_na(*p),
+                uv: Vec2::new(p.x, p.y),
+                color: DARKGRAY,
+            })
+            .collect(),
+        indices.iter().flatten().map(|v| *v as u16).collect(),
+    );
+
+    // Macroquad doesn´t support adding normals to vertices, so we'll bake a color into these vertices.
+    // See https://github.com/not-fl3/macroquad/issues/321.
+
+    // Compute the normal of each vertex, making them unique
+    let vertices: Vec<Vertex> = mquad_compute_normals(&mquad_points, &mquad_indices, camera_pos);
+    // Regenerate the index for each vertex.
+    let indices: Vec<u16> = (0..vertices.len() * 3)
+        .into_iter()
+        .map(|i| i as u16)
+        .collect();
+    let mesh = Mesh {
+        vertices,
+        indices,
+        texture: None,
+    };
+    mesh
+}
+
+fn mquad_compute_normals(points: &Vec<Vertex>, indices: &Vec<u16>, cam_pos: Vec3) -> Vec<Vertex> {
+    let mut vertices: Vec<Vertex> = Vec::<Vertex>::new();
+    for indices in indices.chunks(3) {
+        let v0 = &points[indices[0] as usize];
+        let v1 = &points[indices[1] as usize];
+        let v2 = &points[indices[2] as usize];
+        let normal = (v0.position - v2.position)
+            .cross(v1.position - v2.position)
+            .normalize();
+        let brightness_mod = 0.2 + (0.8 / 2.) * (normal.dot(cam_pos) + 1.);
+
+        for &i in indices.iter() {
+            let mut color = points[i as usize].color;
+            color.r *= brightness_mod;
+            color.g *= brightness_mod;
+            color.b *= brightness_mod;
+
+            vertices.push(Vertex {
+                position: points[i as usize].position,
+                uv: Vec2::ZERO,
+                color: color,
+            });
+        }
+    }
+    vertices
+}
```

Comment:
all this is lengthy, but unfortunately there's not much way around it (see https://github.com/not-fl3/macroquad/issues/321); it would be good to settle on a "common" library for all examples, but as it's the first 3d example this might be ok to keep it here, and we'll move it out with the next example, probably https://github.com/dimforge/parry/pull/250